CharlieBackend.Business/Services/CourseService.cs

```diff
@@ -51,6 +52,10 @@ public async Task<CourseDto> UpdateCourseAsync(long id, UpdateCourseDto courseMo
         {
             try
             {
+                if (await _unitOfWork.CourseRepository.IsCourseEmptyAsync(id))
+                {
+                    return null;
+                }
```

Comment:
new versin of Course Service return Result<CourseDto>, so there might be some conflicts with null